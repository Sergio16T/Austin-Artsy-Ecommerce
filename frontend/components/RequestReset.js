import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { StyledForm, StyledFormWrapper } from "./styles/FormStyles"; 
import Spinner from "./Spinner"; 

const REQUEST_RESET_MUTATION = gql`
    mutation REQUEST_RESET_MUTATION($email: String!) {
        requestReset(email: $email) {
          message
        }
    }
`; 

class RequestReset extends Component {
    state = {
        email: ''
    }
    handleInput = (e) => {
        const {name, value} = e.target; 
        this.setState({
            [name]: value
        }); 
    }
    render() {
        return (
			<StyledFormWrapper>
				<div className="formContainer">
					<Mutation
					mutation={REQUEST_RESET_MUTATION}
					variables={this.state}
					>
					{(reset, {error, loading, called}) => { 
						return (  
						<StyledForm 
						method="post"
						data-test="form" 
						onSubmit={async (e)=> {
							e.preventDefault(); 
							this.setState({spinner: true }); 
							const success = await reset(); 
							console.log(success); 
							this.setState({
									email: '', 
									spinner: false
							}); 
						}}>
							<Spinner spinner={this.state.spinner}/>
							<fieldset disabled={loading} aria-busy={loading}>
								<h2>Request a password reset</h2>
								{!error && !loading && called && <p>Success! Check 
										your email for a reset link</p>}
								<label htmlFor="email">
										Email
										<input 
										type="email" 
										name="email" 
										placeholder="Email" 
										value={this.state.email} 
										onChange={this.handleInput}
										required
										/>
								</label>
								<button type="submit"> Request Reset!</button>
							</fieldset>
						</StyledForm>)
					}}
					</Mutation>
				</div>
			</StyledFormWrapper>
        );
    }
}

export default RequestReset;
export { REQUEST_RESET_MUTATION }; 