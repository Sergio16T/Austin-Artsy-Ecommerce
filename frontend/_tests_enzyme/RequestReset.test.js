import { mount } from 'enzyme'; 
import wait from 'waait'; 
import { shallowToJson } from 'enzyme-to-json'; 
import { MockedProvider } from '@apollo/react-testing'; 
import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset'; 

const mocks = [{
        request: {
            query: REQUEST_RESET_MUTATION, 
            variables: { email: 'stapiafikes@gmail.com'}, 
        }, 
        result: {
            data: { requestReset: { message: 'Thank you!', __typename: 'SuccessMessage'}},
        },
    },
];

describe('<RequestReset/>',() => {
    it('renders and matches snapshot', async ()=> {
        const wrapper = mount(
            <MockedProvider>
                <RequestReset/>
            </MockedProvider>
        )
        // console.log(wrapper.debug()); 
        const form = wrapper.find('form[data-test="form"]'); 
        // console.log(form.debug()); 
        expect(shallowToJson(form)).toMatchSnapshot(); 
    }); 
   it('calls the mutation', async()=> {
    const wrapper = mount(
        <MockedProvider mocks={mocks}>
            <RequestReset/>
        </MockedProvider>
    ); 
    //simulate typing email input
    wrapper.find('input').simulate('change',{ target: { name: 'email', value: 'stapiafikes@gmail.com' } } );
    wrapper.find('form').simulate('submit');
    await wait(); 
    wrapper.update(); 
    expect(wrapper.find('p').text()).toContain('Success!'); 
   }); 
});