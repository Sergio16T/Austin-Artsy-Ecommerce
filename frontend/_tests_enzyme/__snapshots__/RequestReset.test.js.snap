// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="FormStyles__StyledForm-txoayt-1 lkSerF"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <Spinner>
    <Spinner__StyledSpinner>
      <div
        className="Spinner__StyledSpinner-ma7w1f-0 FULCn"
      >
        <div
          id="backDrop"
        >
          <div
            id="loading"
          />
        </div>
      </div>
    </Spinner__StyledSpinner>
  </Spinner>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a password reset
    </h2>
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
       Request Reset!
    </button>
  </fieldset>
</form>
`;
